dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/engine/command/impl/SubmitDatasetForReviewCommand.java

```diff
@@ -58,15 +61,14 @@ public Dataset save(CommandContext ctxt)  {
         updateDatasetUser(ctxt);
 
         AuthenticatedUser requestor = getUser().isAuthenticated() ? (AuthenticatedUser) getUser() : null;
+        List<AuthenticatedUser> curators = ctxt.permissions().getUsersWithPermissionOn(Permission.PublishDataset, savedDataset);
 
-        List<AuthenticatedUser> authUsers = ctxt.permissions().getUsersWithPermissionOn(Permission.PublishDataset, savedDataset);
-        for (AuthenticatedUser au : authUsers) {
-
+        for (AuthenticatedUser au : curators) {
             Option.of(requestor)
                     .peek(user -> ctxt.notifications().sendNotificationWithEmail(au, new Timestamp(new Date().getTime()), NotificationType.SUBMITTEDDS,
-                                                                                 savedDataset.getLatestVersion().getId(), NotificationObjectType.DATASET_VERSION, requestor))
+                                                                                 savedDataset.getLatestVersion().getId(), NotificationObjectType.DATASET_VERSION, requestor, comment))
                     .onEmpty(() -> ctxt.notifications().sendNotificationWithEmail(au, new Timestamp(new Date().getTime()), NotificationType.SUBMITTEDDS,
-                                                                                  savedDataset.getLatestVersion().getId(), NotificationObjectType.DATASET_VERSION));
+                                                                                  savedDataset.getLatestVersion().getId(), NotificationObjectType.DATASET_VERSION, comment));
```

Comment:
Can user who is not authenticated send comment?

Reply:
Old code was prepared for a not-authenticated user to be able to submit dataset for review so I didn't want to change that thinking there may be a business logic that I'm not aware of. Maybe there isn't but I'm not certain.